lib/logstash/plugin_mixins/ecs_compatibility_support/target_check.rb

```diff
@@ -0,0 +1,55 @@
+# encoding: utf-8
+
+require_relative '../ecs_compatibility_support'
+
+module LogStash
+  module PluginMixins
+    module ECSCompatibilitySupport
+      # A target option check that can be included into any `LogStash::Plugin`.
+      #
+      # @see ECSCompatibilitySupport()
+      module TargetCheck
+        ##
+        # @api internal (use: `LogStash::Plugin::include`)
+        # @param base [Class]: a class that inherits `LogStash::Plugin`, typically one
+        #                      descending from one of the four plugin base classes
+        #                      (e.g., `LogStash::Inputs::Base`)
+        # @return [void]
+        def self.included(base)
+          fail(ArgumentError, "`#{base}` must inherit LogStash::Plugin") unless base < LogStash::Plugin
+          fail(ArgumentError, "`#{base}` must include #{ECSCompatibilitySupport}") unless base < ECSCompatibilitySupport
+          base.prepend(RegisterDecorator)
+        end
+
+        TARGET_NOT_SET_MESSAGE = ("ECS compatibility is enabled but no `target` option was specified, " +
+            "it is recommended to set the option to avoid potential schema conflicts (if your data is ECS compliant " +
+            "or non-conflicting feel free to ignore this message)").freeze
+
+        private
+
+        ##
+        # Check whether `target` is specified.
+        #
+        # @return [nil] if ECS compatibility is disabled
+        # @return [true, false]
+        def target_set?
+          return if ecs_compatibility == :disabled
+          ! target.nil?
+        end
```

Comment:
Naming:

As named, I would expect this to return a truthy value when `target` is set, and a falsy value when it is not set, but it returns `nil` when ecs is disabled, whether or not the value is set.

I think we have two options to make this clearer:

1. return `true` when set, and either `false` or `nil` depending on whether ecs is enabled or not
   ```suggestion
           ##
           # Check whether `target` is specified.
           #
           # @return [nil] if target is unspecified and ECScompatibility is disabled
           # @return [true, false]
           def target_set?
             return true unless target.nil?
             return nil if ecs_compatibility == :disabled
             
             false
           end
   ```
2. name it something different that doesn't imply its return value will be truthy IFF the target is set
   ```suggestion
           ##
           # Check whether `target` is encouraged and also specified
           #
           # @return [nil] if ECS compatibility is disabled
           # @return [true, false]
           def target_check?
             return if ecs_compatibility == :disabled
             ! target.nil?
           end
   ```

Reply:
Thanks, have updated to use the first version (keeping `target_set?`), also kind of like the second `target_check?` name given the module is `TargetCheck` ... let me know if you have a preference towards the second one (happy to update if it looks and feels "nicer").